### PR TITLE
libstd needs update for pending libc change

### LINKED ISSUE #39868: libstd needs update for pending libc change
As described in https://github.com/rust-lang/libc/issues/522 , the Solaris use of getpwuid_r is currently incorrect and needs updating to use the standard POSIX interfaces.

I will provide a pull request shortly that depends on pull request https://github.com/rust-lang/libc/issues/523

### PR DESCRIPTION
**Pull request by binarycrusader**
This  updates libstd to accommodate the fixes made in https://github.com/rust-lang/libc/pull/523

Fixes #39868

### CODE REVIEW

**Comment by binarycrusader** [comment]
Please note that this pull request is dependent upon the changes made to libc and cannot be integrated before that one.

File: src/libstd/sys/unix/os.rs [hunk 1]
```diff
@@ -483,7 +483,6 @@ pub fn home_dir() -> Option<PathBuf> {
                   target_os = "nacl",
                   target_os = "emscripten")))]
     unsafe fn fallback() -> Option<OsString> {
-        #[cfg(not(target_os = "solaris"))]
         unsafe fn getpwduid_r(me: libc::uid_t, passwd: &mut libc::passwd,
```
Comment (alexcrichton) [inline src/libstd/sys/unix/os.rs]: Now that there's no abstraction, could this wrapper be removed entirely?

Comment (binarycrusader) [inline src/libstd/sys/unix/os.rs]: Ah, you're right.  I was attempting to minimise changes here, but the wrapper could be removed entirely.  Let me simplify this further and rebuild.

**Comment by binarycrusader** [comment]
Ok, I've simplified this as requested and verified that it works as expected (on Solaris).

**Comment by est31** [comment]
Hmm, shouldn't the libc submodule be updated? Or does this PR update it and github doesn't show it?

**Comment by binarycrusader** [comment]
@est31 This pr does not update the submodule; 1) because I'm new to git and have no idea what the right way is to do that 2) I wasn't certain what rust's policy is for updating submodules.  That's why I explicitly noted in the pull request that this change depends upon the changes made in libc.  My apologies if there is something more that I should do.

If someone would explain to me exactly what I should be doing for pull requests like this, I'm happy to make the requested change.

**Comment by alexcrichton** [comment]
@bors: r-

Oh good catch @est31! @binarycrusader I don't blame you, I try to stay away from submodules with a 10 foot pole... In any case it's totally fine to update the libc submodule whenever, so you're more than welcome to do so. I believe the easiest way to do so is:

```
$ cd src/liblibc
$ git fetch origin
$ git reset --hard origin/master
$ cd ../..
$ git add src/liblibc
$ git commit -m 'Update src/liblibc'
```

**Comment by est31** [comment]
To be honest, I'd have had no idea how to do this myself either. I'm using git since some time already, but I still don't know how to deal with submodules :)

**Comment by binarycrusader** [comment]
Looks like one of the checks failed due to a network error but everything else passed?

**Comment by binarycrusader** [comment]
Can someone get bors to retry the Travis build?  Looks like failure was network or Travis related.

**Pull request merged**